src/objects/core/zcl_abapgit_serialize.clas.abap

```diff
@@ -326,6 +335,56 @@ CLASS zcl_abapgit_serialize IMPLEMENTATION.
   ENDMETHOD.
 
 
+  METHOD filter_unsupported_objects.
+
+    DATA: ls_unsupported_count TYPE ty_unsupported_count,
+          lt_supported_types   TYPE zcl_abapgit_objects=>ty_types_tt,
+          lt_unsupported_count TYPE ty_unsupported_count_tt.
+
+    FIELD-SYMBOLS: <ls_tadir>             LIKE LINE OF ct_tadir,
+                   <ls_unsupported_count> TYPE ty_unsupported_count.
+
+    lt_supported_types = zcl_abapgit_objects=>supported_list( ).
+    LOOP AT ct_tadir ASSIGNING <ls_tadir>.
+      CLEAR: ls_unsupported_count.
+      READ TABLE lt_supported_types WITH KEY table_line = <ls_tadir>-object TRANSPORTING NO FIELDS.
+      IF sy-subrc = 0.
+        CONTINUE.
+      ENDIF.
+
+      READ TABLE lt_unsupported_count ASSIGNING <ls_unsupported_count>
+                                      WITH TABLE KEY obj_type = <ls_tadir>-object.
+      IF sy-subrc <> 0.
+        ls_unsupported_count-obj_type = <ls_tadir>-object.
+        ls_unsupported_count-count    = 1.
+        ls_unsupported_count-obj_name = <ls_tadir>-obj_name.
+        INSERT ls_unsupported_count INTO TABLE lt_unsupported_count ASSIGNING <ls_unsupported_count>.
+      ELSE.
+        CLEAR: <ls_unsupported_count>-obj_name.
+        <ls_unsupported_count>-count = <ls_unsupported_count>-count + 1.
+      ENDIF.
+      CLEAR: <ls_tadir>-object.
+    ENDLOOP.
+    IF lt_unsupported_count IS INITIAL.
+      RETURN.
+    ENDIF.
+
+    DELETE ct_tadir WHERE object IS INITIAL.
+    IF mi_log IS BOUND.
+      LOOP AT lt_unsupported_count ASSIGNING <ls_unsupported_count>.
+        IF <ls_unsupported_count>-count = 1.
+          mi_log->add_error( iv_msg  = |Object type { <ls_unsupported_count>-obj_type } not supported, {
+                                       <ls_unsupported_count>-obj_name } ignored| ).
+        ELSE.
+          mi_log->add_error( iv_msg  = |Object type { <ls_unsupported_count>-obj_type } not supported, {
+                                       <ls_unsupported_count>-count } object(s) ignored| ).
```

Comment:
change "object(s)" to "objects" since now it's always > 1 ;-)

Reply:
changes made